code/src/main/java/com/onlinecontacttracing/servlets/get-data/GetNegativeUserInfoServlet.java

```diff
@@ -0,0 +1,48 @@
+package com.onlinecontacttracing.authentication;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import static com.googlecode.objectify.ObjectifyService.ofy;
+import com.onlinecontacttracing.storage.NegativeUser;
+import java.util.Optional;
+
+@WebServlet("/get-negative-user-info")
+public class GetNegativeUserInfoServlet extends CheckForApiAuthorizationServlet {
+
+  String getServletURIName() {
+    return "/get-negative-user-info";
+  }
+  
+  void useCredential(Credential credential, HttpServletResponse response) throws IOException, InterruptedException {
+    Thread contactInfo = new Thread(new GetCalendarDataForNegativeUser(credential));
+
+    contactInfo.start();
```

Comment:
Why exactly are you kicking off a thread than immediately joining it?

I'm guessing this is an attempt to spin off a thread and then wait for it to finish?

Reply:
And then the data isn't used?

Reply:
The data is stored.

Also this is an attempt at modularity, right now we could do it without a runnable but we could only handle one API, if later we wanted to handle two APIs, for example the Drive API, we would have to make two runnables and move all the code into the runnable.